ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/ReadsKeySets.hs

```diff
@@ -114,6 +118,29 @@ withHydratedLedgerState st dbch urs f =
 -- | The requested point is not found on the ledger db
 data PointNotFound blk = PointNotFound !(Point blk) deriving (Eq, Show)
 
+getLedgerTablesAtFor ::
+     ( HeaderHash l ~ HeaderHash blk
+     , HasHeader blk
+     , IsLedger l
+     , HasTickedLedgerTables l
+     , IOLike m
+     , StandardHash l
+     )
+  => Point blk
+  -> LedgerTables l KeysMK
+  -> LedgerDB l
+  -> LedgerBackingStore m l
+  -> m (Either (PointNotFound blk) (LedgerTables l ValuesMK))
+getLedgerTablesAtFor pt keys lgrDb lbs =
+  case rollback pt lgrDb of
+    Nothing -> pure $ Left $ PointNotFound pt
+    Just l  -> do
+      eValues <-
+        getLedgerTablesFor l keys (readKeySets lbs)
+      case eValues of
+        Right v -> pure $ Right v
+        Left _  -> getLedgerTablesAtFor pt keys lgrDb lbs
```

Comment:
INFO: Previously, this logic was in `LgrDB.getLedgerTablesAtFor`, but it is generally useful to have it here (specially for setting up the ledger interface in the mempool benchmarks)